File: frontend/src/app/service/comentario.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Comentario, ComentarioRequest } from "./ponto-turistico.service";

@Injectable({
  providedIn: 'root'
})

export class ComentarioService {

  private baseUrl = 'http://localhost:8080/api/pontos-turisticos';

  constructor(private http: HttpClient) {}

  //busca comentario por id
  getComentariosByPontoTuristico(pontoTuristicoId: number): Observable<Comentario[]> {
    return this.http.get<Comentario[]>(`${this.baseUrl}/${pontoTuristicoId}/comentarios`);
  }

  //cria um novo comentario
  criarComentario(pontoTuristicoId: number, request: ComentarioRequest): Observable<Comentario> {
    return this.http.post<Comentario>(`${this.baseUrl}/${pontoTuristicoId}/comentarios`, request);
  }

  //deleta comentario
  deletarComentario(pontoTuristicoId: number, comentarioId: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${pontoTuristicoId}/comentarios/${comentarioId}`)
  }
}

File: Backend/src/main/java/br/com/juridico/totvs/fullstack/Backend/service/ComentarioService.java
package br.com.juridico.totvs.fullstack.Backend.service;

import br.com.juridico.totvs.fullstack.Backend.service.dto.ComentarioResponse;
import br.com.juridico.totvs.fullstack.Backend.service.dto.ComentarioRequest;

import java.util.List;

public interface ComentarioService{

    //Cria um comentario para um pponto turistico especifico
    /*
   @param
   @return
     */
    ComentarioResponse criar(ComentarioRequest request);

    //Busca todos os comentarios do ponto turistico
    /*
   @param
   @return
     */
    List<ComentarioResponse> buscarTodosPorPontoTuristico(Long pontoTuristicoId);

    //Deletar um comentario por id
    /*
   @param
   */
    void deletar(Long id);
}
